File: 2025-1/sesion-de-clase-18/SoftProgNet/SoftProgPersistencia/DAOImpl/Logistica/Ventas/LineaOrdenVentaDAOImpl.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.Logistica.Almacen;
using PUCP.Edu.Pe.SoftProg.Modelo.Logistica.Ventas;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.Logistica.Ventas;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.Logistica.Almacen;

namespace PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.Logistica.Ventas {
    public class LineaOrdenVentaDAOImpl : BaseDAOImpl<LineaOrdenVenta>, ILineaOrdenVentaDAO {
        protected override DbCommand CommandoInsertar(DbConnection conn, LineaOrdenVenta lineaOrdenVenta) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "insertarLineaOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idOrdenVenta", DbType.Int32, lineaOrdenVenta.OrdenVenta.Id);
            this.AgregarParametroEntrada(cmd, "@p_idProducto", DbType.Int32, lineaOrdenVenta.Producto.Id);
            this.AgregarParametroEntrada(cmd, "@p_cantidad", DbType.Int32, lineaOrdenVenta.Cantidad);
            this.AgregarParametroEntrada(cmd, "@p_subTotal", DbType.Double, lineaOrdenVenta.SubTotal);
            this.AgregarParametroEntrada(cmd, "@p_activo", DbType.Boolean, lineaOrdenVenta.IsActive);
            this.AgregarParametroSalida(cmd, "@p_id", DbType.Int32);

            return cmd;
        }

        protected override DbCommand CommandoInsertar(DbConnection conn, DbTransaction transaccion, LineaOrdenVenta lineaOrdenVenta) {
            DbCommand cmd = this.CommandoInsertar(conn, lineaOrdenVenta);
            cmd.Transaction = transaccion;

            return cmd;
        }

        protected override DbCommand CommandoModificar(DbConnection conn, LineaOrdenVenta lineaOrdenVenta) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "modificarLineaOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idOrdenVenta", DbType.Int32, lineaOrdenVenta.OrdenVenta.Id);
            this.AgregarParametroEntrada(cmd, "@p_idProducto", DbType.Int32, lineaOrdenVenta.Producto.Id);
            this.AgregarParametroEntrada(cmd, "@p_cantidad", DbType.Int32, lineaOrdenVenta.Cantidad);
            this.AgregarParametroEntrada(cmd, "@p_subTotal", DbType.Double, lineaOrdenVenta.IsActive);
            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, lineaOrdenVenta.Id);

            return cmd;
        }

        protected override DbCommand CommandoEliminar(DbConnection conn, int id) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "eliminarLineaOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand CommandoBuscar(DbConnection conn, int id) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "buscarLineaOrdenVentaPorId";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand CommandoListar(DbConnection conn) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "listarLineasOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;
            return cmd;
        }

        protected override LineaOrdenVenta mapearModelo(DbDataReader reader) {
            return new LineaOrdenVenta {
                Id = Convert.ToInt32(reader["id"]),
                OrdenVenta = new OrdenVentaDAOImpl().Buscar(Convert.ToInt32(reader["idOrdenVenta"])), 
                Producto = new ProductoDAOImpl().Buscar(Convert.ToInt32(reader["idProducto"])), 
                Cantidad = Convert.ToInt32(reader["cantidad"]), 
                SubTotal = Convert.ToDouble(reader["subTotal"]), 
                IsActive = Convert.ToBoolean(reader["activo"])
            };
        }
    }
}

File: 2025-1/sesion-de-clase-20/SoftProg/SoftProgPersistencia/src/main/java/pe/edu/pucp/inf30/softprog/daoimpl/logistica/ventas/LineaOrdenVentaDAOImpl.java
package pe.edu.pucp.inf30.softprog.daoimpl.logistica.ventas;

import java.sql.CallableStatement;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Types;
import pe.edu.pucp.inf30.softprog.dao.logistica.ventas.ILineaOrdenVentaDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.BaseDAOImpl;
import pe.edu.pucp.inf30.softprog.daoimpl.logistica.almacen.ProductoDAOImpl;
import pe.edu.pucp.inf30.softprog.model.logistica.ventas.LineaOrdenVenta;

/**
 *
 * @author eric
 */
public class LineaOrdenVentaDAOImpl extends BaseDAOImpl<LineaOrdenVenta> implements ILineaOrdenVentaDAO {
    @Override
    protected CallableStatement comandoInsertar(Connection conn, LineaOrdenVenta modelo) throws SQLException {
        String sql = "{CALL insertarLineaOrdenVenta(?, ?, ?, ?, ?, ?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt("p_idOrdenVenta", modelo.getOrdenVenta().getId());
        cmd.setInt("p_idProducto", modelo.getProducto().getId());
        cmd.setInt("p_cantidad", modelo.getCantidad());
        cmd.setDouble("p_subTotal", modelo.getSubTotal());
        cmd.setBoolean("p_activo", modelo.isActivo());
        cmd.registerOutParameter("p_id", Types.INTEGER);
        return cmd;
    }

    @Override
    protected CallableStatement comandoModificar(Connection conn, LineaOrdenVenta modelo) throws SQLException {
        String sql = "{CALL modificarLineaOrdenVenta(?, ?, ?, ?, ?, ?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt("p_idOrdenVenta", modelo.getOrdenVenta().getId());
        cmd.setInt("p_idProducto", modelo.getProducto().getId());
        cmd.setInt("p_cantidad", modelo.getCantidad());
        cmd.setDouble("p_subTotal", modelo.getSubTotal());
        cmd.setBoolean("p_activo", modelo.isActivo());
        cmd.setInt("p_id", modelo.getId());
        return cmd;
    }

    @Override
    protected CallableStatement comandoEliminar(Connection conn, int id) throws SQLException {
        String sql = "{CALL eliminarLineaOrdenVenta(?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt(1, id);
        return cmd;
    }

    @Override
    protected CallableStatement comandoBuscar(Connection conn, int id) throws SQLException {
        String sql = "{CALL buscarLineaOrdenVentaPorId(?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt(1, id);
        return cmd;
    }

    @Override
    protected CallableStatement comandoListar(Connection conn) throws SQLException {
        String sql = "{CALL listarLineasOrdenVenta()}";
        CallableStatement cmd = conn.prepareCall(sql);
        return cmd;
    }

    @Override
    protected LineaOrdenVenta mapearModelo(ResultSet rs) throws SQLException {
        LineaOrdenVenta lineaOrdenVenta = new LineaOrdenVenta();
        lineaOrdenVenta.setId(rs.getInt("id"));
        lineaOrdenVenta.setOrdenVenta(new OrdenVentaDAOImpl().buscar(rs.getInt("idOrdenVenta")));
        lineaOrdenVenta.setProducto(new ProductoDAOImpl().buscar(rs.getInt("idProducto")));
        lineaOrdenVenta.setCantidad(rs.getInt("cantidad"));
        lineaOrdenVenta.setSubTotal(rs.getDouble("subTotal"));
        lineaOrdenVenta.setActivo(rs.getBoolean("activo"));
        return lineaOrdenVenta;
    }
}
